packages/core/src/core.ts

```diff
@@ -199,17 +231,49 @@ export function debug(message: string): void {
 /**
  * Adds an error issue
  * @param message error issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation.
  */
-export function error(message: string | Error): void {
-  issue('error', message instanceof Error ? message.toString() : message)
+export function error(
+  message: string | Error,
+  properties: AnnotationProperties = {}
+): void {
+  issueCommand(
+    'error',
+    toCommandProperties(properties),
+    message instanceof Error ? message.toString() : message
+  )
 }
 
 /**
- * Adds an warning issue
+ * Adds a warning issue
  * @param message warning issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation.
  */
-export function warning(message: string | Error): void {
-  issue('warning', message instanceof Error ? message.toString() : message)
+export function warning(
+  message: string | Error,
+  properties: AnnotationProperties = {}
+): void {
+  issueCommand(
+    'warning',
+    toCommandProperties(properties),
+    message instanceof Error ? message.toString() : message
+  )
+}
+
+/**
+ * Adds a notice issue
+ * @param message notice issue message. Errors will be converted to string via toString()
+ * @param properties optional properties to add to the annotation.
+ */
+export function notice(
```

Comment:
can you update the core readme with this function and add some more information about the overloads for column/line and such